pallets/amm/src/mock.rs

```diff
@@ -154,13 +155,24 @@ impl orml_currencies::Config for Test {
 
 parameter_types! {
     pub const AMMPalletId: PalletId = PalletId(*b"par/ammp");
+    pub const AllowPermissionlessPoolCreation: bool = true;
+      pub const AllowPermissionedPoolCreation: bool = false;
 }
 
-impl pallet_amm::Config for Test {
+impl pallet_amm::Config<pallet_amm::Instance1> for Test {
     type Event = Event;
     type Currency = Currencies;
     type PalletId = AMMPalletId;
     type WeightInfo = ();
+    type AllowPermissionlessPoolCreation = AllowPermissionlessPoolCreation;
+}
+
+impl pallet_amm::Config<pallet_amm::Instance2> for Test {
+    type Event = Event;
+    type Currency = Currencies;
+    type PalletId = AMMPalletId;
```

Comment:
I am thinking that to avoid confusions later on, it would be good to have a different `PalletId` here